### PR TITLE
Allow coverage to be displayed for focused specs

### PR DESCRIPTION
**Pull request by joefitzgerald**
In #205, there was lamentation! Show coverage for focused specs, we could not 😢. With this PR, it is possible to opt-in to allow coverage to be shown for focused specs.

The default behavior (failing the suite if any focusing is present, to prevent it from getting through CI) is preserved. A savvy editor integrator can transparently set the `GINKGO_EDITOR_INTEGRATION` environment variable to some non-empty value (e.g. `true`) to suppress the `197` exit code, and allow a coverage file to be generated (and thus allow coverage to be displayed in the editor).

### CODE REVIEW

File: ginkgo/run_command.go [hunk 1]
```diff
@@ -5,14 +5,16 @@ import (
 	"fmt"
 	"math/rand"
 	"os"
+	"strings"
 	"time"
 
+	"io/ioutil"
```
Comment (joefitzgerald) [inline ginkgo/run_command.go]: `goimports` doing it's thing. I'm happy to revert these lines if you'd prefer...

**Comment by jasonkeene** [comment]
I'd ask for the environment variable to be slightly more generic. Perhaps just EDITOR_INTEGRATION=true This will make it easier to get into vim-go as I don't think they want to special case ginkgo.

**Comment by joefitzgerald** [comment]
This is a very ginkgo specific thing. I would imagine a non-specific thing like `EDITOR_INTEGRATION` would be harder and not easier to gain acceptance? I actually thought I might need to be even more specific (e.g. `GINKGO_FOCUSED_COVERAGE`.

File: ginkgo/run_command.go (hunk 1 above)
Comment (onsi) [inline ginkgo/run_command.go]: no problem :)

**Comment by onsi** [comment]
i don't feel too strongly about what to call the env variable - perhaps get input from the vim-go maintainers?

@joefitzgerald can you add an integration test under `/integration` that asserts the exit code is zero for a suite with a focused test when the env variable is passed through?  thanks!

**Comment by joefitzgerald** [comment]
@onsi Integration test added.

**Comment by onsi** [comment]
Sweet thanks!
